### PR TITLE
Migrate dev to google cloud

### PR DESCRIPTION
**Pull request by miikahyttinen**
Added a GH action to deploy  backend to Google Run. Action is triggered on push to `dev` branch.

### CODE REVIEW

File: .github/workflows/deploy-to-dev.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      #  uses: docker/metadata-action@v4
+      #  with:
+      #    images: ${{ env.REGISTRY }}/${{ env.IMAGE }}
+      #    tags: |
+      #      type=raw,value=sha-${{ github.sha }}
+      #    labels: |
+      #      org.opencontainers.image.authors=epoc
+      #      org.opencontainers.image.vendor=three.consulting
+      #    flavor: |
+      #      latest=true
```
Comment (miikahyttinen) [inline .github/workflows/deploy-to-dev.yaml:56]: @LeadingMoominExpert do we need this?

Comment (LeadingMoominExpert) [inline .github/workflows/deploy-to-dev.yaml:56]:
```suggestion
      - name: Define image meta
        id: docker_meta
        uses: docker/metadata-action@v4
        with:
          images: ${{ env.REGION }}-docker.pkg.dev/${{ env.PROJECT_ID }}/${{ env.REPOSITORY }}/${{ env.SERVICE }}
          tags: |
            type=raw,value=${{ github.sha }}
          labels: |
            org.opencontainers.image.authors=epoc
            org.opencontainers.image.vendor=three.consulting
          flavor: |
            latest=true
```

File: .github/workflows/deploy-to-dev.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGION }}-docker.pkg.dev
+          username: 'oauth2accesstoken'
+          password: ${{ steps.auth.outputs.access_token }}
+
+      - name: Build and Push Container
+        run: |-
+          docker build -t "${{ env.REGION }}-docker.pkg.dev/${{ env.PROJECT_ID }}/${{ env.REPOSITORY }}/${{ env.SERVICE }}:${{ github.sha }}" ./
+          docker push "${{ env.REGION }}-docker.pkg.dev/${{ env.PROJECT_ID }}/${{ env.REPOSITORY }}/${{ env.SERVICE }}:${{ github.sha }}"
```
Comment (LeadingMoominExpert) [inline .github/workflows/deploy-to-dev.yaml:76]:
```suggestion
      - name: Build and push image to artifact registry
        uses: docker/build-push-action@v4
        with:
          context: .
          file: Dockerfile
          no-cache: false
          push: true
          tags: ${{ steps.docker_meta.outputs.tags }}
          labels: ${{ steps.docker_meta.outputs.labels }}
```

**Review by LeadingMoominExpert — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-to-dev.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+on:
+  push:
+    branches:
+      - dev
+  workflow_dispatch:
+
+env:
+  PROJECT_ID: epoc-auth-dev-361109
+  FAR_LOCATION: europe-north1
```
Comment (LeadingMoominExpert) [inline .github/workflows/deploy-to-dev.yaml:11]: `FAR_LOCATION` ei oo käytössä

**Review by LeadingMoominExpert — APPROVED** [review]
💯

**Pull request merged**